test/Test.re

```diff
@@ -1,1061 +1,716 @@
+        RemoteAction.act(rAction, ~action=BoxList.Create("Hello"));
+        let (rendered1, _) as actual1 =
+          RenderedElement.flushPendingUpdates(rendered0);
+        RemoteAction.act(rAction, ~action=BoxList.Create("World"));
+        let (rendered2, _) as actual2 =
+          RenderedElement.flushPendingUpdates(rendered1);
+        RemoteAction.act(rAction, ~action=BoxList.Reverse);
+        let (rendered3, _) as actual3 =
+          RenderedElement.flushPendingUpdates(rendered2);
+        TestRenderer.convertElement(rendered0)
+        |> Assert.check(
+             renderedElement,
+             "Initial BoxList",
+             [TestComponents.(<BoxList id=1 />)]
+           );
+        assertFlushUpdate(
+          ~label="Add Hello then Flush",
+          TestComponents.(
+            [<BoxList id=1> <Box id=2 state="Hello" /> </BoxList>],
+            [
+              UpdateInstance({
+                stateChanged: true,
+                subTreeChanged:
+                  `ReplaceElements(([], [<Box id=2 state="Hello" />])),
+                oldInstance: <BoxList id=1 />,
+                newInstance:
+                  <BoxList id=1> <Box id=2 state="Hello" /> </BoxList>
+              })
+            ]
+          ),
+          actual1
         );
-      let expected = [
-        {
-          component: Component(component),
-          id: 1,
-          state: "",
-          subtree: [
-            {
-              component: Component(Div.component),
-              id: 2,
-              state: "",
-              subtree: [
-                {
-                  state: "ImABox",
-                  component: Component(Box.component),
-                  id: 4,
-                  subtree: []
-                },
-                {
-                  state: "ImABox",
-                  component: Component(Box.component),
-                  id: 5,
-                  subtree: []
-                }
-              ]
-            }
-          ]
-        }
-      ];
-      Alcotest.check(
-        renderedElement,
-        "",
-        expected,
-        TestRenderer.convertElement(rendered)
-      )
-    }
-  ),
-  (
-    "Change counter test",
-    `Quick,
-    () => {
-      ReasonReact.GlobalState.reset();
-      let rendered0 =
-        ReasonReact.RenderedElement.render(
-          <ChangeCounter label="defaultText" />
+        assertFlushUpdate(
+          ~label="Add Hello then Flush",
+          TestComponents.(
+            [
+              <BoxList id=1>
+                <Box id=3 state="World" />
+                <Box id=4 state="Hello" />
+              </BoxList>
+            ],
+            [
+              UpdateInstance({
+                stateChanged: true,
+                subTreeChanged:
+                  `ReplaceElements((
+                    [<Box id=2 state="Hello" />],
+                    [<Box id=3 state="World" />, <Box id=4 state="Hello" />]
```

Comment:
Shouldn't this be `PrependElement`? If not, why? Actually, it shouldn't be, but why not? We need to document this.